### DISCUSSION TITLE
Is there any technical limitations by using .setState in components?

### DISCUSSION THREAD

**Discussion by joelbqz** [pmndrs/zustand#951] [Q&A] [answered]
Hey! I'm wondering If I'll run into any issues with this approach? The main reason of this is I want to have full type inference:
```
export const useState = create(() => {
  return {
    counter: 0,
  };
});

const { setState: set, getState: get } = useState;

const actions = {
  increase: () => set({ counter: get().counter + 1 }),
};
```

[Quick demo](https://stackblitz.com/edit/react-ts-jdu6l3?file=App.tsx)

**Comment by dai-shi** [comment, marked as answer]
No.
See also: https://github.com/pmndrs/zustand/wiki/Practice-with-no-store-actions
